sway-ir/tests/tests.rs

```diff
@@ -72,8 +78,23 @@ fn inline() {
 fn constants() {
     run_tests("constants", |ir: &mut sway_ir::Context| {
         let fn_idcs: Vec<_> = ir.functions.iter().map(|func| func.0).collect();
-        fn_idcs.into_iter().all(|fn_idx| {
+        fn_idcs.into_iter().fold(false, |acc, fn_idx| {
             sway_ir::optimize::combine_constants(ir, &sway_ir::function::Function(fn_idx)).unwrap()
+                || acc
+        })
+    })
+}
+
+// -------------------------------------------------------------------------------------------------
+
+#[allow(clippy::needless_collect)]
+#[test]
+fn simplify_cfg() {
+    run_tests("simplify_cfg", |ir: &mut sway_ir::Context| {
+        let fn_idcs: Vec<_> = ir.functions.iter().map(|func| func.0).collect();
+        fn_idcs.into_iter().fold(false, |acc, fn_idx| {
```

Comment:
Do we need to collect here? cant we keep the iterator lazy?

Reply:
Borrow checker says No.